### PR TITLE
Feat - Add Note Tags

### PR DESCRIPTION
**Pull request by reddhilario**
noteable changes: 

- added tag creation with color randomizer

### CODE REVIEW

File: src/assets/main.css [hunk 1]
```diff
@@ -32,4 +32,123 @@ a,
...
+  background: rgb(120, 222, 172);
+}
+
+.pastel-cotton-candy {
+  background: rgb(255, 188, 217);
+}
+
+.pastel-silver {
+  background: rgb(192, 192, 192);
+}
```
Comment (ezekielreginio) [inline src/assets/main.css]: try to declare these colors css as constants reference:https://stackoverflow.com/questions/4847850/is-it-possible-to-define-constants-in-css

File: src/router/index.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import {
   createRouter,
   createWebHistory
 } from 'vue-router'
-import HomeView from '../views/HomeView.vue'
+// import HomeView from '../views/HomeView.vue'
```
Comment (ezekielreginio) [inline src/router/index.js]: remove if unused

File: src/views/todo/TodoView.vue [hunk 3]
```diff
@@ -18,10 +18,21 @@ import SideNavigationComponent from './subcomponents/SideNavigationComponent.vue
...
+  },
   components: {
     TopNavigationComponent,
     SideNavigationComponent,
     TaskContainerComponent
+  },
+  methods: {
+    toggleSideBarOpen() {
+      this.sideBarOpen = !this.sideBarOpen
+      console.log('Parent Component: ' + this.sideBarOpen)
```
Comment (ezekielreginio) [inline src/views/todo/TodoView.vue]: remove console logs

File: src/views/todo/subcomponents/MainComponent.vue [hunk 4]
```diff
@@ -1,58 +1,218 @@
...
+              id="tagDropDownButton"
+              data-bs-toggle="dropdown"
+              aria-expanded="false"
+            ></button>
+            <ul class="dropdown-menu" aria-labelledby="tagDropDownButton">
+              <li class="dropdown-item">
+                <div class="rounded-pill bg-warning" style="height: 10px; width: 30px"></div>
+              </li>
+            </ul>
+          </div> -->
```
Comment (ezekielreginio) [inline src/views/todo/subcomponents/MainComponent.vue]: remove if unused

File: src/views/todo/subcomponents/MainComponent.vue [hunk 5]
```diff
@@ -1,58 +1,218 @@
...
+      <div class="shadow rounded p-1" id="addTask">
+        <div class="input-group input-group-sm">
+          <input
+            placeholder="Title"
+            type="text"
+            class="form-control border-0"
+            aria-label="Sizing example input"
+            aria-describedby="inputGroup-sizing-sm"
+            @input="setTitle($event.target.value)"
+          />
```
Comment (ezekielreginio) [inline src/views/todo/subcomponents/MainComponent.vue]: class as last attrib

File: src/views/todo/subcomponents/MainComponent.vue [hunk 6]
```diff
@@ -1,58 +1,218 @@
...
+        </div>
+        <hr class="m-0" />
+        <div class="input-group">
+          <textarea
+            :style="{ height: this.textAreaHeight }"
+            ref="textArea"
+            placeholder="Take a note..."
+            class="form-control border-0"
+            aria-label="With textarea"
+            @input="setDescription($event.target.value)"
```
Comment (ezekielreginio) [inline src/views/todo/subcomponents/MainComponent.vue]: class as last attrib

File: src/views/todo/subcomponents/MainComponent.vue (hunk 5 above)
Comment (ezekielreginio) [reply src/views/todo/subcomponents/MainComponent.vue]: add ID tag for input field

File: src/views/todo/subcomponents/MainComponent.vue (hunk 6 above)
Comment (ezekielreginio) [reply src/views/todo/subcomponents/MainComponent.vue]: add ID tag for input field

File: src/views/todo/subcomponents/MainComponent.vue [hunk 7]
```diff
@@ -1,58 +1,218 @@
...
+          <TagComponent v-for="(listedTag, index) in listedTags" :key="index" :tag="listedTag" />
+          <div class="dropdown">
+            <button
+              class="btndropdown-toggle rounded-pill pb-2 border-0"
+              type="button"
+              id="tagDropDownButton"
+              data-bs-toggle="dropdown"
+              aria-expanded="false"
+              style="height: auto; width: auto"
+            >
```
Comment (ezekielreginio) [inline src/views/todo/subcomponents/MainComponent.vue]: class as last attrib

File: src/views/todo/subcomponents/MainComponent.vue [hunk 8]
```diff
@@ -1,58 +1,218 @@
...
+              style="width: 300px; height: 200px; overflow: auto; background: rgb(68, 76, 92)"
+            >
+              <input
+                class="w-100 border border-secondary border-end-0 border-start-0 border-top-0"
+                type="text"
+                style="overflow: hidden; background: transparent; font-size: small; color: white"
+                placeholder="Search or Create"
+                @input="setTagName($event.target.value)"
+                @keyup.enter="createTag"
+              />
```
Comment (ezekielreginio) [inline src/views/todo/subcomponents/MainComponent.vue]: class and style as last attrib. Add ID

File: src/views/todo/subcomponents/MainComponent.vue [hunk 9]
```diff
@@ -61,31 +221,96 @@ export default {
...
+    },
+    randomColor() {
+      //   console.log(
+      //     '' +
+      //       JSON.stringify(this.colors[Math.floor(Math.random() * this.colors.length)]).substring(
+      //         1,
+      //         JSON.stringify(this.colors[Math.floor(Math.random() * this.colors.length)]).length + 1
+      //       )
+      //   )
+      //   return this.colors[Math.random() * this.colors.length]
```
Comment (ezekielreginio) [inline src/views/todo/subcomponents/MainComponent.vue]: remove if unused

File: src/views/todo/subcomponents/MainComponent.vue [hunk 10]
```diff
@@ -61,31 +221,96 @@ export default {
...
+      console.log('' + typeof this.colors[Math.floor(Math.random() * this.colors.length)])
```
Comment (ezekielreginio) [inline src/views/todo/subcomponents/MainComponent.vue]: remove console logs

**Review by ezekielreginio — CHANGES_REQUESTED** [review]
12 comments

File: src/assets/main.css (hunk 1 above)
Comment (reddhilario) [reply src/assets/main.css]: finished, happy?

**Review by ezekielreginio — APPROVED** [review]
12 comments

**Pull request merged**